fixbackend/auth/auth_backend.py

```diff
@@ -0,0 +1,129 @@
+#  Copyright (c) 2023. Some Engineering
+#  This program is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU Affero General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU Affero General Public License for more details.
+#
+#  You should have received a copy of the GNU Affero General Public License
+#  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import hashlib
+from datetime import datetime, timedelta
+from typing import Any, Dict, List, Optional
+from uuid import UUID
+
+import jwt
+from cryptography.hazmat.primitives import serialization
+from cryptography.hazmat.primitives.asymmetric.rsa import RSAPrivateKey, RSAPublicKey
+from fastapi_users import exceptions
+from fastapi_users.authentication import AuthenticationBackend, JWTStrategy
+from fastapi_users.authentication.strategy.base import Strategy, StrategyDestroyNotSupportedError
+from fastapi_users.manager import BaseUserManager
+
+from fixbackend.auth.models import User
+from fixbackend.auth.transport import CookieTransport
+from fixbackend.certificates.cert_store import CertificateStoreDependency
+from fixbackend.config import ConfigDependency
+
+
+class FixJWTStrategy(Strategy[User, UUID]):
+    def __init__(
+        self,
+        public_key_1: RSAPublicKey,
+        public_key_2: RSAPublicKey,
```

Comment:
The strategy itself should handle a list of keys. It is an implementation detail, that we only use 2.